File: brainbox/min_operations_sum_divisible_by_k.py
def min_operations(nums, k):
    total = sum(nums)
    return total % k


# -------------------------
# Test Cases
# -------------------------

tests = [
    # Provided example
    ([3, 9, 7], 5, 4),

    # Already divisible
    ([1, 2, 3], 3, 0),

    # Large numbers
    ([100, 200, 300], 7, (100+200+300) % 7),

    # Single element
    ([10], 6, 10 % 6),

    # Needs only 1 decrement
    ([4, 4], 5, (4+4) % 5),

    # All zeros
    ([0, 0, 0], 4, 0),

    # Mixed positives
    ([5, 11, 14], 6, (5+11+14) % 6),
]

for nums, k, expected in tests:
    result = min_operations(nums, k)
    print(f"nums={nums}, k={k} → {result} (expected {expected})")

File: brainbox/min_operations_sum_divisible_by_k.ts

// -------------------------
// Solution
// -------------------------
export function minOperations(nums: number[], k: number): number {
  const sum = nums.reduce((a, b) => a + b, 0);
  return sum % k;
}

// -------------------------
// Tests
// -------------------------
const tests: { nums: number[]; k: number; expected: number }[] = [
  { nums: [3, 9, 7], k: 5, expected: 4 },
  { nums: [1, 2, 3], k: 3, expected: 0 },
  { nums: [100, 200, 300], k: 7, expected: (100 + 200 + 300) % 7 },
  { nums: [10], k: 6, expected: 10 % 6 },
  { nums: [4, 4], k: 5, expected: (4 + 4) % 5 },
  { nums: [0, 0, 0], k: 4, expected: 0 },
  { nums: [5, 11, 14], k: 6, expected: (5 + 11 + 14) % 6 },
];

tests.forEach(({ nums, k, expected }) => {
  const result = minOperations(nums, k);
  console.log(`nums=${JSON.stringify(nums)}, k=${k} → ${result} (expected ${expected})`);
});
